### PR TITLE
Fix bugs in rnd

### PR DESCRIPTION
**Pull request by matt439**
# Description

The `rnd(int ubound)` and `rnd(int min, int max)` overloads of `rand` generate arithmetic exceptions when a divide by zero occurs through a modulo operator. `rnd(0)` will reliably generate the exception. When `min = max`, the divide by zero will occur in `rnd(int min, int max)` as `... % (max - min)` is used in the function.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

I added calls to rnd(0) and rnd(10, 10) to `test_graphics.cpp` to test the edge cases. While it may seem out of place to use `rnd` in `test_graphics.cpp`, `#include "random.h"` was already present in the file with no associated references. Now the `#include` is justified.

## Testing Checklist

- [x] Tested with sktest

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

File: coresdk/src/coresdk/random.cpp [hunk 1]
```diff
@@ -32,6 +34,8 @@ namespace splashkit_lib
     
     int rnd(int min, int max)
     {
+        if (min == max) return min;
```
Comment (Liquidscroll) [inline coresdk/src/coresdk/random.cpp:37]: What if min is greater than max? For example running `rnd(50, 10)` returned 56. Should this print an error if this occurs?

File: coresdk/src/test/test_graphics.cpp [hunk 2]
```diff
@@ -26,7 +26,8 @@ void test_drawing_on_new_window()
     save_bitmap(user_image, "0");
     clear_bitmap (user_image, COLOR_BRIGHT_GREEN);
     save_bitmap(user_image, "1");
-    fill_rectangle_on_bitmap (user_image, COLOR_BLACK, 0, 0, 10, 10);
+    // rnd(0) should return 0, and rnd(10, 10) should return 10
+    fill_rectangle_on_bitmap (user_image, COLOR_BLACK, rnd(0), 0, rnd(10, 10), 10);
```
Comment (Liquidscroll) [inline coresdk/src/test/test_graphics.cpp:30]: I don't think this is a great place for this; this should be testing graphics only. The `rnd` functions seem like they'd be a good for unit testing.

**Review by Liquidscroll — CHANGES_REQUESTED** [review]
Good work! A few issues: 

- What if the user passes in a value for `min` that is greater than `max`? 
- I'm not a fan of the mixing of functionality in the graphics test, this might be better in its own unit test.

EDIT: I just saw PR#68 adds the unit tests for `rnd()`

File: coresdk/src/coresdk/random.cpp (hunk 1 above)
Comment (matt439) [reply coresdk/src/coresdk/random.cpp:44]: I added a check for this edge case. I see three potential actions: throw an exception, return 0, or return rnd(max, min). I believe the third option is the best as most people who call rnd(20, 10) are expecting it to behave as rnd(10, 20) anyway. I added an output to the console whenever the function is used improperly. Throwing an exception seems extreme in this case and returning 0 could confuse users.

File: coresdk/src/test/test_graphics.cpp (hunk 2 above)
Comment (matt439) [reply coresdk/src/test/test_graphics.cpp]: I agree with this. I was concerned that this specific pull request would be deemed insufficient if there was no testing included. I reverted the changes to `test_graphics.cpp`.

File: coresdk/src/coresdk/random.cpp (hunk 1 above)
Comment (Liquidscroll) [reply coresdk/src/coresdk/random.cpp:44]: Perhaps adding some form of clarification to this string would be good. Something to let the users know that we'll be swapping the values they've input into the function.

I don't think this should block this from the next review, so I'll approve this.

**Review by Liquidscroll — APPROVED** [review]
Great job! Left a comment about a clarification that could be made; but I don't think it should block this from the next review.

File: coresdk/src/coresdk/random.cpp (hunk 1 above)
Comment (hugglesfox) [reply coresdk/src/coresdk/random.cpp:44]: The correct answer imo would be to throw an error anytime the function is being used incorrectly. However as splashkit is scared of throwing exceptions as they're too scary for people new to programming, I think this is an acceptable way of handling the error.

**Review by hugglesfox — APPROVED** [review]
lgtm

**Review by WhyPenguins — CHANGES_REQUESTED** [review]
Looks good! Just would be better to make the `rnd` error be output via the in-built logging system - after that this should be good to merge.

File: coresdk/src/coresdk/random.cpp [hunk 3]
```diff
@@ -32,6 +35,14 @@ namespace splashkit_lib
     
     int rnd(int min, int max)
     {
+        if (min > max)
+        {
+            std::cout << "Error: min value is greater than max value when calling rnd" << std::endl;
```
Comment (WhyPenguins) [inline coresdk/src/coresdk/random.cpp:40]: Just for consistency this should probably be output via `LOG(WARNING)` rather than `cout`

Comment (matt439) [reply coresdk/src/coresdk/random.cpp]: I have altered the output to use LOG(WARNING). It required me to include easylogging++.h in the file.
